src/Microsoft.DotNet.Darc/Darc/Helpers/GitHubCliTokenProvider.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using Maestro.Common;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+namespace Microsoft.DotNet.Darc.Helpers;
+
+/// <summary>
+/// Token provider that attempts to retrieve a GitHub token from the GitHub CLI ('gh auth token')
+/// if no static token is provided. Falls back to the provided token or null if gh CLI is not available.
```

Comment:
nit: I don't think it falls back to the provided token, if gh cli is not available, it's the other way around, cause we check the provided token first